glib bindings: linking fails on OS X, FreeBSD

On both systems gobject-introspection is successfully found
```
-- Checking for module 'gobject-2.0>=2.36'
--   Found gobject-2.0, version 2.46.2
-- Include Binding glib
-- Checking for module 'gobject-introspection-1.0>=1.38'
--   Found gobject-introspection-1.0, version 1.46.0
```

Fails on both systems very similarly.
```
[ 72%] Linking C shared library ../../../lib/libgelektra-4.0.dylib
cd /Users/mpranj/workspace/libelektra/build/src/bindings/glib && /usr/local/Cellar/cmake/3.5.2/bin/cmake -E cmake_link_script CMakeFiles/gelektra-4.0.dir/link.txt --verbose=1
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc   -std=gnu99  -Wno-deprecated-declarations  -Wno-long-long -pedantic -Wno-variadic-macros -Wall -Wextra -Wno-overlength-strings -Wsign-compare -Wfloat-equal -Wformat-security -Wshadow -Wcomments -Wtrigraphs -Wundef -Wuninitialized -Winit-self -Wsign-compare -Wfloat-equal -Wformat-security -O2 -g -DNDEBUG -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk -dynamiclib -Wl,-headerpad_max_install_names  -o ../../../lib/libgelektra-4.0.dylib -install_name @rpath/libgelektra-4.0.dylib CMakeFiles/gelektra-4.0.dir/gelektra-kdb.c.o CMakeFiles/gelektra-4.0.dir/gelektra-key.c.o CMakeFiles/gelektra-4.0.dir/gelektra-keyset.c.o ../../../lib/libelektra-kdb.0.8.16.dylib -lgobject-2.0 -lglib-2.0 -lintl ../../../lib/libelektra-core.0.8.16.dylib -Wl,-rpath,/Users/mpranj/workspace/libelektra/build/lib
ld: library not found for -lgobject-2.0
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [lib/libgelektra-4.0.dylib] Error 1
```

--- a/src/bindings/glib/CMakeLists.txt
+++ b/src/bindings/glib/CMakeLists.txt
@@ -1,5 +1,12 @@
 include (LibAddMacros)
 
+if (CMAKE_MAJOR_VERSION EQUAL 2 AND CMAKE_MINOR_VERSION EQUAL 8)
+	if (CMAKE_PATCH_VERSION LESS 12)
+		remove_binding(jna "cmake 2.8.12 needed for glib")
+		return()
+	endif()
+endif()
+
 find_package (PkgConfig)
 pkg_check_modules (GOBJECT gobject-2.0>=2.36)
 

--- a/src/bindings/glib/CMakeLists.txt
+++ b/src/bindings/glib/CMakeLists.txt
@@ -12,18 +12,18 @@ else()
 
 	file (GLOB GELEKTRA_HDR_FILES *.h)
 	file (GLOB GELEKTRA_SRC_FILES *.c)
-
 	add_headers (ELEKTRA_HEADERS)
 	set (SOURCES ${GELEKTRA_SRC_FILES} ${GELEKTRA_HDR_FILES} ${ELEKTRA_HEADERS})
 
-	include_directories (${CMAKE_CURRENT_SOURCE_DIR})
-
-	add_definitions (${GOBJECT_CFLAGS})
-
 	add_library (${GELEKTRA_LIBRARY} SHARED ${SOURCES})
 
-	target_link_libraries (${GELEKTRA_LIBRARY} elektra-core elektra-kdb)
-	target_link_libraries (${GELEKTRA_LIBRARY} ${GOBJECT_LIBRARIES})
+	target_include_directories (${GELEKTRA_LIBRARY} PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
+	target_include_directories (${GELEKTRA_LIBRARY} SYSTEM PUBLIC ${GOBJECT_INCLUDE_DIRS})
+
+	target_compile_options (${GELEKTRA_LIBRARY} PUBLIC ${GOBJECT_CFLAGS_OTHER})
+
+	target_link_libraries (${GELEKTRA_LIBRARY} PUBLIC elektra-core elektra-kdb)
+	target_link_libraries (${GELEKTRA_LIBRARY} PUBLIC ${GOBJECT_LDFLAGS})
 
 	install (
 		TARGETS ${GELEKTRA_LIBRARY}

--- a/src/bindings/glib/tests/CMakeLists.txt
+++ b/src/bindings/glib/tests/CMakeLists.txt
@@ -1,5 +1,3 @@
-include_directories (${CMAKE_SOURCE_DIR}/tests)
-
 set (TEST_FILES $<TARGET_OBJECTS:cframework>)
 add_testheaders(TEST_FILES)
 
@@ -14,9 +12,7 @@ macro (do_glib_test file)
 		COMPILE_DEFINITIONS HAVE_KDBCONFIG_H
 	)
 
-	set (GLIBS ${GELEKTRA_LIBRARY} ${GOBJECT_LIBRARIES})
-	tool_link_elektra (${name})
-	target_link_libraries (${name} ${GLIBS})
+	target_link_libraries (${name} PRIVATE ${GELEKTRA_LIBRARY})
 
 	add_test (
 		NAME ${name}

